File: ZooJava/src/main/java/com/example/zoojava/controllers/LoginController.java
package com.example.zoojava.controllers;

import com.example.zoojava.DI.components.DaggerEmployeeRepositoryComponent;
import com.example.zoojava.managers.SceneManager;
import com.example.zoojava.repositories.employees.EmployeesRepository;
import com.example.zoojava.utils.Globals;
import javafx.event.ActionEvent;
import javafx.fxml.FXML;
import javafx.scene.control.*;
import javafx.scene.image.ImageView;
import javafx.stage.Stage;



/**
 * Controlador del login
 */
public class LoginController {
    private EmployeesRepository repository= DaggerEmployeeRepositoryComponent.create().build();
    @FXML
    private ImageView imagenLogo;
    @FXML
    private TextField emailText;
    @FXML
    private TextField passwordText;



    /**
     * Para cerrar el programa, dependiendo de su confirmación.
     */
    public void closeProgram() {
        Alert alerta = new Alert(Alert.AlertType.CONFIRMATION);
        alerta.setTitle("Cerrar programa");
        alerta.setHeaderText("Desea cerrar el programa?");
        alerta.setContentText("Si acepta se cerrará el programa");
        var respuesta = alerta.showAndWait();

        if (respuesta.get()== ButtonType.OK){
            Stage stage = (Stage) imagenLogo.getScene().getWindow();
            stage.close();
        }
    }


    /**
     * Para abrir el acerca de, del programa.
     */
    public void openAbout() {
        SceneManager.openAbout();
    }


    /**
     * Login del usuario.
     */
    public void login(ActionEvent actionEvent) {
        if (!isErrorsEmpty()){
            isCorrectLogin();
        }
    }


    /**
     * Para saber si el usuario es correcto.
     */
    private void isCorrectLogin() {
        var user = repository.findByEmail(emailText.getText());
            if(user!=null){
                if (user.getPassword().equals(passwordText.getText())){
                    Globals.globalEmployee=user;
                    Alert alerta = new Alert(Alert.AlertType.INFORMATION);
                    alerta.setTitle("Login Correcto");
                    alerta.setHeaderText("Bienvenido "+ user.getName());
                    alerta.show();
                } else{
                    Alert alert = new Alert(Alert.AlertType.ERROR);
                    alert.setTitle("Error de login");
                    alert.setContentText("Login incorrecto");
                    alert.show();
                }
            }else{
                Alert alert = new Alert(Alert.AlertType.ERROR);
                alert.setTitle("Error de login");
                alert.setContentText("Usuario no encontrado");
                alert.show();
            }
    }


    /**
     * Método para ver si el usuario se ha dejado campos vacíos.
     * @return boolean dependiendo de si ha habido errores o no.
     */
    private boolean isErrorsEmpty() {
        StringBuilder sb = new StringBuilder();
        if (emailText.getText().equals("") || passwordText.getText().equals("")){
            sb.append("No se pueden dejar campos vacíos").append("\n");
            Alert alert = new Alert(Alert.AlertType.ERROR);
            alert.setTitle("Error de campos");
            alert.setContentText(sb.toString());
            alert.show();
            return true;
        }
       return false;
    }


}

File: ZooKotlin/src/main/kotlin/com/example/zookotlin/controllers/LoginController.kt
package com.example.zookotlin.controllers

import com.example.zookotlin.managers.SceneManager
import com.example.zookotlin.repositories.employees.EmployeesRepository
import com.example.zookotlin.repositories.employees.EmployeesRepositoryImpl
import com.example.zookotlin.utils.Globals
import javafx.event.ActionEvent
import javafx.fxml.FXML
import javafx.scene.control.Alert
import javafx.scene.control.ButtonType
import javafx.scene.control.TextField
import javafx.scene.image.ImageView
import javafx.stage.Stage

/**
 * Controlador del login.
 */
class LoginController {
    private val repository: EmployeesRepository = EmployeesRepositoryImpl.getInstance()!!

    @FXML
    private lateinit var imagenLogo: ImageView

    @FXML
    private lateinit var emailText: TextField

    @FXML
    private lateinit var passwordText: TextField


    /**
     * Para cerrar el programa, dependiendo de su confirmación.
     */
    fun closeProgram() {
        val alerta = Alert(Alert.AlertType.CONFIRMATION)
        alerta.title = "Cerrar programa"
        alerta.headerText = "Desea cerrar el programa?"
        alerta.contentText = "Si acepta se cerrará el programa"
        val respuesta = alerta.showAndWait()
        if (respuesta.get() == ButtonType.OK) {
            val stage = imagenLogo.scene.window as Stage
            stage.close()
        }
    }


    /**
     * Para abrir el acerca de, del programa.
     */
    fun openAbout() {
        SceneManager.openAbout()
    }


    /**
     * Login del usuario.
     */
    fun login() {
        if (!isErrorsEmpty()) {
            isCorrectLogin()
        }
    }


    /**
     * Para saber si el usuario es correcto.
     */
    private fun isCorrectLogin() {
        val user = repository.findByEmail(emailText.text)
        if (user != null) {
            if (user.getPassword() == passwordText.text) {
                Globals.globalEmployee = user
                val alerta = Alert(Alert.AlertType.INFORMATION)
                alerta.title = "Login Correcto"
                alerta.headerText = "Bienvenido " + user.getName()
                alerta.show()
                SceneManager.openInicio(imagenLogo.scene.window as Stage)
            } else {
                val alert = Alert(Alert.AlertType.ERROR)
                alert.title = "Error de login"
                alert.contentText = "Login incorrecto"
                alert.show()
            }
        } else {
            val alert = Alert(Alert.AlertType.ERROR)
            alert.title = "Error de login"
            alert.contentText = "Usuario no encontrado"
            alert.show()
        }
    }


    /**
     * Método para ver si el usuario se ha dejado campos vacíos.
     * @return boolean dependiendo de si ha habido errores o no.
     */
    private fun isErrorsEmpty(): Boolean {
        val sb = StringBuilder()
        if (emailText.text == "" || passwordText.text == "") {
            sb.append("No se pueden dejar campos vacíos").append("\n")
            val alert = Alert(Alert.AlertType.ERROR)
            alert.title = "Error de campos"
            alert.contentText = sb.toString()
            alert.show()
            return true
        }
        return false
    }

}